forge/src/runner.rs

```diff
@@ -268,8 +274,50 @@ impl<'a, DB: DatabaseRef + Send + Sync> ContractRunner<'a, DB> {
     ) -> Result<SuiteResult> {
         tracing::info!("starting tests");
         let start = Instant::now();
+        let mut warnings = Vec::new();
         let needs_setup = self.contract.functions().any(|func| func.name == "setUp");
 
+        let setup_fns: Vec<_> = self
+            .contract
+            .functions()
+            .filter_map(|func| {
+                if func.name.to_lowercase() == "setup" {
+                    Some(func.signature())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        // There are multiple setUp function, so we return a single test result for `setUp`
+        if setup_fns.len() > 1 {
+            return Ok(SuiteResult::new(
+                start.elapsed(),
+                [(
+                    "setUp()".to_string(),
+                    TestResult {
+                        success: false,
+                        reason: Some("Multiple setUp functions".to_string()),
+                        counterexample: None,
+                        logs: vec![],
+                        kind: TestKind::Standard(0),
+                        traces: vec![],
+                        labeled_addresses: BTreeMap::new(),
+                    },
+                )]
+                .into(),
+                warnings,
```

Comment:
Maybe we should push a warning for each of the setup functions?

Reply:
Ahh, just realised this currently will only ever get called if there is one setUp method miss-cased. Otherwise the error will return before. Maybe I should push the `setUp` warnings before I return the error? Then it will be super clear for the user. What do you think?

Reply:
I did this, seems to make sense to have the warnings even when it errors. Gives some extra context about what bad setUp functions where found.